src/libexpr/primops/fetchTree.cc

```diff
@@ -74,8 +74,11 @@ std::string fixURI(std::string uri, EvalState & state, const std::string & defau
 
 std::string fixURIForGit(std::string uri, EvalState & state)
 {
+    /* Detects scp-style uris (e.g. git@github.com:NixOS/nix) and fixes
+     * them by removing the `:` and assuming a scheme of `ssh://`
+     * */
     static std::regex scp_uri("([^/].*)@(.*):(.*)");
```

Comment:
Couldn’t this be fixed by just changing the regex to

```suggestion
    static std::regex scp_uri("([^/]*)@(.*):(.*)");
```

?

I have the feeling that this `.` was just a typo

Reply:
(I think that the best specification for what’s allowed in an ssh username is https://datatracker.ietf.org/doc/html/rfc2396#section-3.2.2 , which fobids `/` in a username. So it’s not overly restrictive either)

Reply:
:bulb: that is true. Let me quickly fix that